client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -160,34 +161,32 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 			return null;
 		}
 
-		const firstAuthWalledStep = flowSteps.find( ( step ) => step.requiresLoggedInUser );
-
-		if ( step.slug === 'user' && firstAuthWalledStep ) {
+		// The `nextStep` is available only when logged-out users go to the step that requires auth
+		// and are redirected to the user step.
+		const nextStepSlug = stepData?.nextStep ?? '';
```

Comment:
I am thinking that maybe there's a better name for `nextStepSlug`, as it seems too generic for the specific case that the comments are explaining.

What do you think of `postAuthStepSlug`?